libexec/pyenv-init

```diff
@@ -297,9 +297,15 @@ EOS
   fi
 
   case "\$command" in
+EOS
+  if [ ${#commands[@]} -ne 0 ]; then
+    cat <<EOS
   ${commands[*]})
```

Comment:
I think the fix can be simplified to

```suggestion
  "${commands[@]}")
```

Reply:
Scratch that. That would match the absense of a command argument -- which is valid input.
Still, it rubs me the wrong way to add loads of code for a case that never happens in practice. Would very much prefer smth more naturally flowing.

Reply:
I'm not really a fan of this code, either, but it's the best I could come up with to avoid difficult-to-debug errors caused by `grep` and `sort` being miscompiled on my system.

An alternative implementation I considered was adding a placeholder command if the array is empty, but as far as I know, a plugin author is free to name custom commands whatever they want, which could cause a conflict.

Reply:
Okay, take 2:

```suggestion
  "${commands[@]:-/}")
```

`/` is the only character in Linux besides `\0` that filenames cannot contain. So a valid command can never be "`/`".

Reply:
Take 3:

```suggestion
  ${commands[*]:-/})
```

This one expands correctly both when the array in empty and non-empty

Reply:
@native-api, I can confirm that works! I have modified `pyenv-init`, the test, and the commit message accordingly.